src/nvim/edit.c

```diff
@@ -1685,7 +1685,7 @@ static void init_prompt(int cmdchar_todo)
   // Insert always starts after the prompt, allow editing text after it.
   if (Insstart_orig.lnum != curwin->w_cursor.lnum || Insstart_orig.col != (colnr_T)STRLEN(prompt)) {
     Insstart.lnum = curwin->w_cursor.lnum;
-    Insstart.col = STRLEN(prompt);
+    Insstart.col = (int)STRLEN(prompt);
```

Comment:
a cast to `colnr_T` is better imo

```suggestion
    Insstart.col = (colnr_T)STRLEN(prompt);
```

Reply:
I guess it makes sense, given that it is more specific.